tools/CouchDB_MRI_Importer.php

```diff
@@ -96,24 +96,82 @@ function _generateCandidatesQuery($ScanTypes)
 
         foreach($s as $scan){
             $scantype=$scan['ScanType'];
-            $Query .= ", (SELECT f.File FROM files f LEFT JOIN mri_scan_type msc
-              ON (msc.ID= f.AcquisitionProtocolID)
-              WHERE f.SessionID=s.ID AND msc.Scan_type='$scantype' LIMIT 1)
-                    as `Selected_$scantype`, (SELECT fqc.QCStatus
-                    FROM files f 
-                    LEFT JOIN files_qcstatus fqc USING(FileID)
-                    LEFT JOIN mri_scan_type msc ON(msc.ID= f.AcquisitionProtocolID)
-              WHERE f.SessionID=s.ID AND msc.Scan_type='$scantype' LIMIT 1)
-                     as `$scantype"."_QCStatus`";
+
+            //-------------------------------------------------------------------------//
+            // Add to the query a subselect that will compute the selected file        //
+            // for the given session/modality. If more than one selected file exists,  //
+            // the subquery will return 'Multiple Selected files'. If no selected file //
+            // exists, the subselect will return 'No selected file'. Otherwise the     //
+            // file path of the selected file is returned by the subselect             //
+            //-------------------------------------------------------------------------//
+            $Query .= ', '
+                    . '('
+                    . ' CASE ('. $this->_getQueryForSelectedFiles('count(*)', $scantype) . ') '
+                    . '  WHEN 1 '
+                    . '    THEN (' . $this->_getQueryForSelectedFiles('f.File', $scantype) . ') '
+                    . '  WHEN 0 '
+                    . '    THEN "No selected file" '
+                    . '  ELSE "Multiple Selected files"'
```

Comment:
Why only showing multiple here and not return the list of filenames? I get that they would not be downloadable anyway but I feel it would help the process to be able to see the filenames of the files listed in that cell when there is more than one file so you can then provide a list of the filenames you wish to get transferred to you?

Reply:
@nicolasbrossard any answers ?

Reply:
@ridz @cmadjar I think displaying all the file names would take up too much space in the result table. I believe we decided to use 'Multiple Selected files' instead in an imaging meeting but I could be wrong...